substrate/frame/broker/src/dispatchable_impls.rs

```diff
@@ -455,4 +454,25 @@ impl<T: Config> Pallet<T> {
 
 		Ok(())
 	}
+
+	pub(crate) fn ensure_cores_for_sale(
+		status: &StatusRecord,
+		sale: &SaleInfoRecordOf<T>,
+	) -> Result<(), DispatchError> {
+		ensure!(sale.first_core < status.core_count, Error::<T>::Unavailable);
+		ensure!(sale.cores_sold < sale.cores_offered, Error::<T>::SoldOut);
+
+		Ok(())
+	}
+
+	/// If there is an ongoing sale returns the current price of a core.
+	pub fn api_sale_price() -> Result<BalanceOf<T>, DispatchError> {
+		let status = Status::<T>::get().ok_or(Error::<T>::Uninitialized)?;
+		let sale = SaleInfo::<T>::get().ok_or(Error::<T>::NoSales)?;
+
+		Self::ensure_cores_for_sale(&status, &sale)?;
+
+		let now = frame_system::Pallet::<T>::block_number();
```

Comment:
When we change to relay chain clocking this could give strange results if not updated. Maybe best to have a simple regression test here.

Reply:
The good thing is that a different type is used to define `now`, so if it isn't updated, it shouldn't even compile.

Reply:
Fair enough!